BEAN-1621: feature: add account to company (#8)
Beans-BV/merchant_sdk_javascript: README.md

      - [Generate Deeplink](#generate-deeplink)
      - [Generate PNG QR Code](#generate-png-qr-code)
      - [Generate SVG QR Code](#generate-svg-qr-code)
      - [Create Company Account](#create-company-account)
      - [Upload Company Account Avatar](#upload-company-account-avatar)
      - [Delete Company Account](#delete-company-account)
      - [Get Company Account Avatar](#get-company-account-avatar)
  - [Webhook Notifications](#webhook-notifications)
- [Questions and Answers](#questions-and-answers)
  - [Do I have to use stroops?](#do-i-have-to-use-stroops)
- [Examples](#examples)
    - [Checkout](#checkout)

>     - [Sub-Account Management](#sub-account-management)

Reviewer: ```suggestion
    - [Account Management](#account-management)
```

  });
```

#### Create Company Account

> *Creates a sub-account for the company.*

Reviewer: ```suggestion
*Creates an account for the company or its clients to receive payments..*
```

Method Signature:<br>
*`Promise<CreateCompanyAccountResponse> createCompanyAccount(...)`*

Parameters:<br>

>   - `stellarAccountId`: *The Stellar account ID for the sub-account.*

Reviewer: ```suggestion
  - `stellarAccountId`: *The Stellar account ID for the account.*
```

>   - `name`: *The name of the sub-account in different languages as a map where the key is the language code (e.g., 'en', 'vi') and the value is the name in that language.*

Reviewer: ```suggestion
  - `name`: *The name of the account in different languages as a map where the key is the language code (e.g., 'en', 'vn') and the value is the name in that language.*
```

Returns:<br>
`Promise<CreateCompanyAccountResponse>`: *A promise that resolves with the response object containing the created company account.*

Return Object Properties:<br>

>   - `account`: *The CompanyAccount object representing the created sub-account.*

Reviewer: ```suggestion
  - `account`: *The CompanyAccount object representing the created account.*
```

Example:<br>
```js
const name = {
  'en': 'Marketing Account',

>   'vi': 'Tài khoản Marketing'

Reviewer: ```suggestion
  'vn': 'Tài khoản Marketing'
```

};

sdk.createCompanyAccount('GBZX4364PEPQTDICMIQDZ56K4T75QZCR4NBEYKO6PDRJAHZKGUOJPCXB', name)
  .then((response) => {
    console.log('Created company account:', response.account.id);
  });
```

#### Upload Company Account Avatar

> *Uploads an avatar for a company sub-account.*

Reviewer: ```suggestion
*Uploads an avatar for a company account.*
```

Method Signature:<br>
*`Promise<CompanyAccount> uploadCompanyAccountAvatar(...)`*

Parameters:<br>

>   - `companyId`: *The ID of the company or 'me' for the current company.*

Reviewer: ```suggestion
  - `companyId`: *The ID of the company, or the string "me" to automatically resolve the ID from the provided API token.*
```

>   - `stellarAccountId`: *The Stellar account ID of the sub-account.*

Reviewer: ```suggestion
  - `stellarAccountId`: *The Stellar account ID of the account.*
```

  - `imageData`: *The image data as a File, Blob, or ArrayBuffer.*

Returns:<br>
`Promise<CompanyAccount>`: *A promise that resolves with the updated CompanyAccount object.*

Example:<br>
```js
// Using a File from file input
const imageFile = document.getElementById('avatarFile').files[0];

sdk.uploadCompanyAccountAvatar('me', 'GBZX4364PEPQTDICMIQDZ56K4T75QZCR4NBEYKO6PDRJAHZKGUOJPCXB', imageFile)
  .then((updatedAccount) => {
    console.log('Account updated with avatar:', updatedAccount.avatarId);
  });
```

#### Delete Company Account

> *Deletes a sub-account for the company.*

Reviewer: ```suggestion
*Deletes an account from the company.*
```

Method Signature:<br>
*`Promise<DeleteCompanyAccountResponse> deleteCompanyAccount(...)`*

Parameters:<br>

>   - `stellarAccountId`: *The Stellar account ID of the sub-account to delete.*

Reviewer: ```suggestion
  - `stellarAccountId`: *The Stellar account ID of the account to delete.*
```

Returns:<br>

> `Promise<DeleteCompanyAccountResponse>`: *A promise that resolves with the response object containing information about the deleted sub-account.*

Reviewer: ```suggestion
`Promise<DeleteCompanyAccountResponse>`: *A promise that resolves with the response object containing information about the deleted account.*
```

Return Object Properties:<br>

>   - `account`: *The CompanyAccount object representing the deleted sub-account.*

Reviewer: ```suggestion
  - `account`: *The CompanyAccount object representing the deleted account.*
```

  - `status`: *The status of the deletion operation, typically "deleted".*

Example:<br>
```js
sdk.deleteCompanyAccount('GCQYCNYU3T73JCQ2J36A3JJ5CUQO4DY4EOKMPUL5723ZH7N6XMMNPAA3')
  .then((response) => {
    console.log('Deleted account:', response.account.id);
    console.log('Status:', response.status);
  });
```

#### Get Company Account Avatar

> *Gets the avatar for a company sub-account.*

Reviewer: ```suggestion
*Gets the avatar for a company account.*
```

Method Signature:<br>
*`Promise<ArrayBuffer> getCompanyAccountAvatar(...)`*

Parameters:<br>

>   - `companyId`: *The ID of the company or 'me' for the current company.*

Reviewer: ```suggestion
  - `companyId`: *The ID of the company, or the string "me" to automatically resolve the ID from the provided API token.*
```

>   - `accountId`: *The ID of the sub-account.*

Reviewer: ```suggestion
  - `accountId`: *The ID of the account.*
```

## Do I have to use stroops?

Our platform utilizes Stellar blockchain technology to simplify digital transactions, making it accessible even to those unfamiliar with cryptocurrencies. We've streamlined the payment process using `Stellar Lumen (XLM)` and other cryptocurrencies, allowing you to transact in main currency units instead of dealing with complex conversions like `stroops`. For example, to request a payment of `1 XLM`, you just set the amoun to `"1"` and our system automatically handles the conversion to `10,000,000 stroops`, ensuring a user-friendly payment experience.

# Examples

We've provided examples to help you understand what you as a business can do with the Beans Merchant SDK.

### Checkout

This example showcases a checkout page that allows users to pay for items using Beans. You can see how easy it is to integrate Beans payments into your e-commerce platform.

> 

Reviewer: ```suggestion

Find the full example code [here](https://github.com/Beans-BV/merchant_sdk_javascript/tree/main/example/shopping_basket).

```

> ### Sub-Account Management

Reviewer: ```suggestion
### Account Management
```

> We've added an example that demonstrates how to create and manage company sub-accounts. This example shows how to:

Reviewer: ```suggestion
We've added an example that demonstrates how to create and manage company accounts. This example shows how to:
```

> - Create a new sub-account with multi-language support

Reviewer: ```suggestion
- Create a new account with multi-language support
```

> - Upload an avatar for a sub-account

Reviewer: ```suggestion
- Upload an avatar for an account
```

- Retrieve and display the avatar

> - Delete a sub-account when it's no longer needed

Reviewer: ```suggestion
- Delete an account when it's no longer needed
```

This functionality is particularly useful for businesses that need to manage multiple Stellar accounts under a single company account.

Find the full example code [here](https://github.com/Beans-BV/merchant_sdk_javascript/blob/main/example/subaccount/index.html).

### Advanced (for developers)

> This example is more technical and aimed at developers who want to understand the full capabilities of the Beans Merchant SDK.

Reviewer: ```suggestion
This example is more technical and aimed at developers who want to understand the full capabilities of the Beans Merchant SDK.

Find the full example code [here](https://github.com/Beans-BV/merchant_sdk_javascript/tree/main/example/shopping_basket).

```

# Examples

We've provided examples to help you understand what you as a business can do with the Beans Merchant SDK.

### Checkout

This example showcases a checkout page that allows users to pay for items using Beans. You can see how easy it is to integrate Beans payments into your e-commerce platform.

Find the full example code [here](https://github.com/Beans-BV/merchant_sdk_javascript/tree/main/example/shopping_basket).

### account Management

We've added an example that demonstrates how to create and manage company accounts. This example shows how to:
- Create a new account with multi-language support
- Upload an avatar for an account
- Retrieve and display the avatar
- Delete an account when it's no longer needed

This functionality is particularly useful for businesses that need to manage multiple Stellar accounts under a single company account.

Find the full example code [here](https://github.com/Beans-BV/merchant_sdk_javascript/blob/main/example/subaccount/index.html).

### Advanced (for developers)

This example is more technical and aimed at developers who want to understand the full capabilities of the Beans Merchant SDK.

> Find the full example code [here](https://github.com/Beans-BV/merchant_sdk_javascript/tree/main/example/shopping_basket).

Reviewer: ```suggestion

Find the full example code [here](https://github.com/Beans-BV/merchant_sdk_javascript/tree/main/example/shopping_basket).
```

## Do I have to use stroops?

Our platform utilizes Stellar blockchain technology to simplify digital transactions, making it accessible even to those unfamiliar with cryptocurrencies. We've streamlined the payment process using `Stellar Lumen (XLM)` and other cryptocurrencies, allowing you to transact in main currency units instead of dealing with complex conversions like `stroops`. For example, to request a payment of `1 XLM`, you just set the amoun to `"1"` and our system automatically handles the conversion to `10,000,000 stroops`, ensuring a user-friendly payment experience.

# Examples

We've provided examples to help you understand what you as a business can do with the Beans Merchant SDK.

### Checkout

This example showcases a checkout page that allows users to pay for items using Beans. You can see how easy it is to integrate Beans payments into your e-commerce platform.

Find the full example code [here](https://github.com/Beans-BV/merchant_sdk_javascript/tree/main/example/shopping_basket).

> 

Reviewer: ```suggestion
```

### account Management

We've added an example that demonstrates how to create and manage company accounts. This example shows how to:
- Create a new account with multi-language support
- Upload an avatar for an account
- Retrieve and display the avatar
- Delete an account when it's no longer needed

This functionality is particularly useful for businesses that need to manage multiple Stellar accounts under a single company account.

Find the full example code [here](https://github.com/Beans-BV/merchant_sdk_javascript/blob/main/example/account/index.html).

### Advanced (for developers)

This example is more technical and aimed at developers who want to understand the full capabilities of the Beans Merchant SDK.

> 

Reviewer: ```suggestion
```
